src/rbac/rbac.js

```diff
@@ -78,8 +78,10 @@ class RBAC {
 
     for (let u in this._users) {
       const user = this.getUser(u);
-      if (user.role === roleName) {
-        this.rmUser(user.name);
+      const roles = user.roles;
+      if (roles.indexOf(roleName) > -1) {
+        const newRoles = roles.filter(role => role !== roleName);
```

Comment:
could save the index in a const and then use `Array.splice` to remove the role, instead of iterating the array twice via `indexOf` and `filter`

Reply:
though this catches duplicates?

Reply:
I think using `splice` and then checking for duplicates elsewhere, such as `addUser`, may be better. Right now, all adding users and editing users both go through `addUser`.

Reply:
Yes that sounds like a good plan to me. Add user and possibly if we can just use `Joi` to enforce it that would be best.